readme.txt
=== Ziggeo Video for Ninja Forms ===
Contributors: oliverfriedmann, baned, carloscsz409, natashacalleia
Tags: ziggeo, video, video field, form builder, video form, Ninja Forms
Requires at least: 3.0.1
Tested up to: 6.1.1
Stable tag: 1.7.1
Requires PHP: 5.2.4
License: GPLv3 or later
License URI: http://www.gnu.org/licenses/gpl-3.0.html

This plugin brings video player and recorder (including screen recording) to your Ninja Forms. It helps you make it easy to add fields that utilize award winning Ziggeo to your forms.

Please note that you need to install and setup [Ziggeo plugin](https://wordpress.org/plugins/ziggeo/) first. This plugin is offered as an extension of the same.

== Who is this for? ==

Do you need to enrich your forms with multimedia?
Want to capture screen recording to help your support team when issues are submitted?
Looking to make your HR team work better and easier by offering camera recordgins and previews?
If any of the above is true, this plugin is what you want to add to your Wordpress website.

= Benefits =

Allows you to quickly add video playback and recording to your forms.
Ninja Forms native implementation of different Ziggeo solutions.
Add player, recorder, screen recorder and more to your forms.
Great support to offer you assistance with setup.

== Screenshots ==

1. Form Fields - Seamless integration if Ziggeo fields with other Ninja Form fields
2. Form Builder - Video Player
3. Form Builder - Video Recorder
4. Form Builder - VideoWall Field
5. Form Builder - Ziggeo Templates
6. Submissions - Video Preview
7. Admin options


== Installation ==
 
1. Upload plugin zip file to your plugins directory. Usually that would be to the `/wp-content/plugins/` directory
1. Activate the plugin through the 'Plugins' menu in WordPress
1. That is it or
1. Use the Plugins Add new section to find the plugin and install
 
== Frequently Asked Questions ==

= How does it work? =

This plugin will provide you with the Ziggeo Fields section in your form editor. Once you open it, it will reveal all of the different types of fields that we support.

By a simple drag and drop you can quickly add multimedia to your forms.

= Where does integration happen? =

Integration happens within your website. All the data you gather will still be available to you in same panels and integrations as before.

As always we will host multimedia that is captured within your Ziggeo account and link to the same will be used as a submitted value on your form.

= How to use Dynamic Custom Data =

Ziggeo internally supports the ability of adding custom data to your videos. This can be anything as long as it is provided as valid JSON field. Now with form builders you might want to add custom data based on the data in the fields as well. To do this, we bring you dynamic custom data field.

* Please note that this field should not be used in combination with the custom data. You should use either `Custom Data` or `Dynamic Custom Data`.

The way you would set it up is by using key:field_id. For example if you want your JSON to be formed as:

[javascript]
{
	"first_name": "Mike",
	"last_name": "Wazowski"
}
[/javascript]

and let's say that your first name has `<input id="nf-field-tmp-2" ...>` and last name has `<input id="nf-field-tmp-3" ...>`. It means that we need `nf-field-tmp-2` and `nf-field-tmp-3` to get those values. So our field can be set as:

`first_name:nf-field-tmp-2,last_name:nf-field-tmp-3`

As you save your recorder field it will remember this and try to find the values. If the fields with ID are not found, the value will be saved as "" (empty string)

= How can I get some support =

We provide active support to all that have any questions or need any assistance with our plugin or our service.
To submit your questions simply go to our [Help Center](https://support.ziggeo.com/hc/en-us). Alternatively just send us an email to [support@ziggeo.com](mailto:support@ziggeo.com).

= I have an idea or suggestion =

Please go to our [WordPress forum](https://support.ziggeo.com/hc/en-us/community/topics/200753347-WordPress-plugin) and add your suggestion within it. This allows everyone to see and vote on it and us to determine what should be next.

== Upgrade Notice ==

= 1.7.1 =
* Improvement: Added additional checks for lazy load to make sure it works fine when core version does not yet support lazy load.

== Changelog ==

Please check CHANGELOG.md found in the plugin files. This file contains all of the log entries for past versions.


CHANGELOG.md
This file contains the change log info for the `Ziggeo - Ninja Forms` Bridge plugin


=======

= 1.7 =
* Added: Support for the WP Core plugin lazyload feature
* Improvement: Submissions can be viewed in the Submissions panel even with Lazy Load feature turned on

= 1.6.2 =
* Fix: In some cases there was an error show in admin panel and this fix prevents it from happening.

= 1.6 =
* Fix: Setting what values to be saved was not working properly. This has been addressed with this new update. Your submissions could be saved in a different manner than default if your settings were set to something else. If this is the case, you can always change them by going to Ziggeo Video > Ziggeo Video for Ninja Forms in your WordPress dashboard.

= 1.5 =
* Improvement: Seems that in some cases Ninja Forms fields get some styles added that can make it harder to click on the upload file button. This update adds the CSS that address the same.

= 1.4 =
* Added: Now you can set the custom data from dynamic values. The only important part is that once verified event fires (after recording) it will try to capture the fields you set. It will not capture any changes made after.
* Added: Added hooks for the Ninja forms verified event, allowing you to add your own code handler to react once video is verified within the Ninja Forms form.
* Modified: Custom (dynamic) tags and capturing token is happening through the new verified event, allowing you to see how to set up and use the same.

= 1.3 =
* Improvement: API calls now use V2 only
* Improvement: Submenu creation now has a check to make sure that the core plugin is installed before calling it's function, which will help with any hidden errors during install if core plugin is not present.
* Improvement: Plugin is now going to report if the core plugin is not available instead of silently skipping the load of files.

= 1.2 =
* Added custom tags (tags based on other fields on the form) which are applied to the video once the video is verified.

= 1.1 =
* Cleanup and integrations updates

= 1.0 =
Initial commit

core/run.php
<?php

//
//	This file represents the integration module for Ninja Forms and Ziggeo
//

// Index
//	1. Hooks
//		1.1. ziggeo_list_integration
//		1.2. plugins_loaded
//	2. Functionality
//		2.1. ziggeoninjaforms_get_version()
//		2.2. ziggeoninjaforms_init()
//		2.3. ziggeoninjaforms_run()

//Checking if WP is running or if this is a direct call..
defined('ABSPATH') or die();

//Show the entry in the integrations panel
add_filter('ziggeo_list_integration', function($integrations) {

	$current = array(
		//This section is related to the plugin that we are combining with the Ziggeo, not the plugin/module that does it
		'integration_title'		=> 'Ninja Forms', //Name of the plugin
		'integration_origin'	=> 'https://ninjaforms.com', //Where you can download it from

		//This section is related to the plugin or module that is making the connection between Ziggeo and the other plugin.
		'title'					=> 'Ziggeo Video for Ninja Forms', //the name of the module
		'author'				=> 'Ziggeo', //the name of the author
		'author_url'			=> 'https://ziggeo.com/', //URL for author website
		'message'				=> 'Add video to form builder and your forms', //Any sort of message to show to customers
		'status'				=> true, //Is it turned on or off?
		'slug'					=> 'ziggeo-video-for-ninja-forms', //slug of the module
		//URL to image (not path). Can be of the original plugin, or the bridge
		'logo'					=> ZIGGEONINJAFORMS_ROOT_URL . 'assets/images/logo.png',
		'version'				=> ZIGGEONINJAFORMS_VERSION
	);

	//Check current Ziggeo version
	if(ziggeoninjaforms_run() === true) {
		$current['status'] = true;
	}
	else {
		$current['status'] = false;
	}

	$integrations[] = $current;

	return $integrations;
});

//add_action('plugins_loaded', function() {
add_action('plugins_loaded', function() {
	ziggeoninjaforms_run();
}, 8);

//Checks if the WPForms exists and returns the version of it
function ziggeoninjaforms_get_version() {

	if(!class_exists('Ninja_Forms')) {
		return 0;
	}

	return Ninja_Forms::VERSION;
}

//Include all of the needed plugin files
function ziggeoninjaforms_include_plugin_files() {

	//Include the files only if we are running this plugin

	include_once(ZIGGEONINJAFORMS_ROOT_PATH . 'core/simplifiers.php');
	include_once(ZIGGEONINJAFORMS_ROOT_PATH . 'core/assets.php');
	include_once(ZIGGEONINJAFORMS_ROOT_PATH . 'core/header.php');
	include_once(ZIGGEONINJAFORMS_ROOT_PATH . 'core/ajax.php');

	//Fields specific
	require_once(ZIGGEONINJAFORMS_ROOT_PATH . 'extend/class-video-recorder.php');
	require_once(ZIGGEONINJAFORMS_ROOT_PATH . 'extend/class-video-player.php');
	require_once(ZIGGEONINJAFORMS_ROOT_PATH . 'extend/class-ziggeo-template.php');


	//Check if there is VideoWalls plugin present or not and include additional field(s) if so
	if(defined('VIDEOWALLSZ_VERSION')) {
		require_once(ZIGGEONINJAFORMS_ROOT_PATH . 'extend/class-videowall.php');
	}
}

//We add all of the hooks we need
function ziggeoninjaforms_init() {

	//Lets include all of the files we need
	ziggeoninjaforms_include_plugin_files();

	//Adding Ziggeo Fields section in the builder
	add_filter( 'ninja_forms_field_type_sections', function($sections) {

		$tmp = array(
			'ziggeo' => array(
				'id'			=> 'ziggeo_fields',
				'nicename'		=> 'Ziggeo Fields',
				'fieldTypes'	=> array()
			)
		);

		$sections = array_slice($sections, 0, 4, true) + $tmp + array_slice($sections, 4, count($sections)-1, true);

		return $sections;
	}, 8);

	//Add the field
	add_filter( 'ninja_forms_register_fields', function ($fields) {
		$fields['ziggeo-video-player'] = new Ninja_Forms_Field_Video_Player();
		$fields['ziggeo-video-recorder'] = new Ninja_Forms_Field_Video_Recorder();
		$fields['ziggeo-template'] = new Ninja_Forms_Field_Ziggeo_Templates();

		if(defined('VIDEOWALLSZ_VERSION')) {
			$fields['videowall'] = new Ninja_Forms_Field_Videowall();
		}

		return $fields;
	});

	//Works on front end when it loads the template
	add_filter( 'ninja_forms_field_template_file_paths', function($paths) {

		//Let us set the path to our templates
		$paths[] = ZIGGEONINJAFORMS_ROOT_PATH . 'templates/';

		return $paths;
	});

	// Detect if lazyload is enabled and if someone is viewing the submissions panel
	if(is_admin() && isset($_GET, $_GET['page']) && $_GET['page'] === 'nf-submissions') {
		if(function_exists('ziggeo_p_is_lazyloaded') && ziggeo_p_is_lazyloaded() === true) {
			// We need to add various things
			if(!defined('ZIGGEO_FOUND')) {
				define('ZIGGEO_FOUND', true);
			}

			echo ziggeo_p_get_lazyload_activator();

			if(!defined('ZIGGEO_FOUND_POST')) {
				define('ZIGGEO_FOUND_POST', true);
			}
		}
	}

}


//Function that we use to run the module 
function ziggeoninjaforms_run() {

	//Needed during activation of the plugin
	if(!function_exists('ziggeo_get_version')) {
		add_action( 'admin_notices', function() {
			?>
			<div class="error notice">
				<p><?php _e( 'Please install <a href="https://wordpress.org/plugins/ziggeo/">Ziggeo plugin</a>. It is required for this plugin (Ziggeo Video for Ninja Forms) to work properly!', 'ziggeoninjaforms' ); ?></p>
			</div>
			<?php
		});

		return false;
	}

	//Check current Ziggeo version
	if( version_compare(ziggeo_get_version(), '2.0') >= 0 &&
		//check the Ninja Forms version
		version_compare(ziggeoninjaforms_get_version(), '3.4.23') >= 0) {

		if(ziggeo_integration_is_enabled('ziggeo-video-for-ninja-forms')) {
			ziggeoninjaforms_init();
			return true;
		}
	}

	return false;
}


?>